### ISSUE TITLE
[Feature] ColorJitterOp (brightness, contrast, saturation, hue)

### ISSUE THREAD

**Issue by brody-0125** [brody-0125/dart_tensor_preprocessing#32]
## Summary

Add `ColorJitterOp` for random brightness, contrast, saturation, and hue adjustments.

## Motivation

Color jitter is one of the most impactful data augmentation techniques for vision models, improving robustness to lighting and color variations. This is a standard component in virtually every modern training pipeline.

## Reference Implementations

- **PyTorch**: `torchvision.transforms.ColorJitter(brightness=0, contrast=0, saturation=0, hue=0)` — [PyTorch docs](https://pytorch.org/vision/stable/generated/torchvision.transforms.ColorJitter.html)
- **TensorFlow**: Separate functions `tf.image.adjust_brightness`, `tf.image.adjust_contrast`, `tf.image.adjust_saturation`, `tf.image.adjust_hue` — [TF docs](https://www.tensorflow.org/api_docs/python/tf/image/adjust_brightness)

## Verification

- ✅ `torchvision.transforms.ColorJitter` exists — VERIFIED
- ✅ `tf.image.adjust_brightness` exists — VERIFIED
- ✅ Feature is MISSING from dart_tensor_preprocessing — VERIFIED (searched for brightness, contrast, saturation, hue, jitter, color)

## Proposed API

```dart
// Combined color jitter (PyTorch-style)
final jitter = ColorJitterOp(
  brightness: 0.2,   // ±20% brightness
  contrast: 0.2,     // ±20% contrast
  saturation: 0.2,   // ±20% saturation
  hue: 0.1,          // ±0.1 hue shift
);

// Individual operations (TensorFlow-style)
final bright = AdjustBrightnessOp(delta: 0.2);
final contrast = AdjustContrastOp(factor: 1.5);
final saturation = AdjustSaturationOp(factor: 1.3);
final hue = AdjustHueOp(delta: 0.1);
```

## Implementation Notes

- Brightness: `pixel + delta` or `pixel * factor`
- Contrast: `(pixel - mean) * factor + mean` per channel
- Saturation: Convert RGB→HSV, multiply S channel, convert back
- Hue: Convert RGB→HSV, rotate H channel, convert back
- Requires RGB↔HSV color space conversion (see related issue)
- Support 3D `[C, H, W]` and 4D `[N, C, H, W]` tensors
- Add random range support (e.g., brightness=(0.8, 1.2))
- Optional seed for reproducibility

## Dependencies

- Color space conversion (RGB↔HSV) — should be implemented first or concurrently

## ROI Assessment

**Very High** — Color jitter is essential for any serious vision training pipeline. Combined with the existing normalization and resize ops, this completes the standard augmentation toolkit.
